### PR TITLE
Bump jsoup to 1.15.3

### PR DESCRIPTION
**Pull request by wzieba**
Version without reported security vulnerabilities

To address https://github.com/wordpress-mobile/WordPress-Android/security/dependabot/91

### Test
Smoke test the "code editor" (last button) in AztecDemo app.

### CODE REVIEW

File: aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt [hunk 1]
```diff
@@ -30,9 +32,10 @@ object Format {
         CleaningUtils.cleanNestedBoldTags(doc)
         if (isCalypsoFormat) {
             // remove empty span tags
-            doc.select("*")
-                    .filter { !it.hasText() && it.tagName() == "span" && it.childNodes().size == 0 }
-                    .forEach { it.remove() }
+            val select: Elements = doc.select("*")
+            select.filter { element: Element ->
```
Comment (ParaskP7) [inline aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt]: Minor (🔍): Why exposing this `select` local variable if not using it, maybe you had something in mind? 🤔 Consider, reverting it to it's previous form: `doc.select("*").filter { element: Element -> ... }`

**Review by ParaskP7 — APPROVED** [review]
👋 @wzieba !

I have reviewed and tested this PR as well, everything works as expected, good job! 🌟 

-----

I have just a minor (🔍) comment for you to consider. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself (after fixing the UI test issue).

File: aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt (hunk 1 above)
Comment (wzieba) [reply aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt]: I thought it'll be more readable, but maybe it's not the case. Inlined in: 5bee1c97376322727875b624ae26048542e608ae

**Comment by ParaskP7** [comment]
> (after fixing the UI test issue)

Btw, did we manage to fix the UI test issue @wzieba ? 🤔

**Comment by wzieba** [comment]
The test happened to be just flaky. I run it locally a few times, it always passed. On CI, I rerun it twice and passed each time.

**Comment by ParaskP7** [comment]
Oh great, thanks for the confirmation on that! 🙇 😌

**Pull request merged**